src/printer.js

```diff
@@ -2843,7 +2843,11 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
     // the other ones since we don't call the normal print on BinaryExpression,
     // only for the left and right parts
     if (isNested && node.comments) {
-      parts.push(comments.printComments(path, p => "", options));
+      parts.splice(
+        0,
+        parts.length,
+        comments.printComments(path, p => concat(parts.slice()), options)
+      );
```

Comment:
I think this code could be a little more readable. Doesn't this do the same thing?

```js
parts = comments.printComments(path, p => concat(parts), optons)
```

?